### ISSUE TITLE
RFC rename snapshot_download.py

### ISSUE THREAD

**Issue by adrinjalali** [huggingface/huggingface_hub#875]
Right now we have `snapshot_download.py` which defines `snapshot_download` as a function. As a result `huggingface_hub.snapshot_download` doesn't have clear semantics, is it the module or the function?

I suggest we rename the file to make the distinction clear.

This issue was surfaced while working on https://github.com/huggingface/huggingface_hub/pull/874

cc @LysandreJik @osanseviero

**Comment by LysandreJik** [comment]
Yes, sound good to me as well. I don't think we want to encourage `from huggingface_hub.snapshot_download import snapshot_download`, so ok for me to rename that file.
